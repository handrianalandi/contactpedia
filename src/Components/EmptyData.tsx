import React from "react";
import emptyDataImage from "../assets/empty-data-vector.png";
import styled from "@emotion/styled";

const EmptyDataWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  height: fit-content;
`;

const EmptyDataText = styled.h5`
  font-size: 1rem;
  font-weight: 700;
  color: grey;
  margin: 0;
  text-align: center;
`;

interface EmptyDataProps {
  text?: string;
}
export default function EmptyData({ text = "No Data Found" }: EmptyDataProps) {
  return (
    <EmptyDataWrapper>
      <img src={emptyDataImage} alt="Empty Data" />
      <EmptyDataText>{text}</EmptyDataText>
    </EmptyDataWrapper>
  );
}
