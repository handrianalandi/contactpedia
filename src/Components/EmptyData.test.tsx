import { render } from "@testing-library/react";
import EmptyData from "./EmptyData";

describe("EmptyData", () => {
  it("should render correctly", () => {
    const { container } = render(<EmptyData />);
    expect(container).toMatchSnapshot();
  });
});
