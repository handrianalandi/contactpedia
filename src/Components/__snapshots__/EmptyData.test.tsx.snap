// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyData should render correctly 1`] = `
<div>
  <div
    class="css-1vlo6q4"
  >
    <img
      alt="Empty Data"
      src="empty-data-vector.png"
    />
    <h5
      class="css-1ak0za4"
    >
      No Data Found
    </h5>
  </div>
</div>
`;
